src/pairs.rs

```diff
@@ -89,6 +93,23 @@ fn rewrite_pairs_one_line<T: Rewrite>(
     wrap_str(result, context.config.max_width(), shape)
 }
 
+fn set_pair_string<'a>(
+    result: &'a mut String,
+    comment: &Option<String>,
+    base_shape: Shape,
+    context: &RewriteContext<'_>,
+) -> &'a String {
+    if let Some(comment) = comment {
+        let indent_str = base_shape.indent.to_string_with_newline(context.config);
```

Comment:
Could you try to keep the original position of the comment, please? E.g.,

```rust
    long_expression_0 // post comment 0
        // pre comment 1
        * long_expression_1 // post commnt 1
        // pre comment 2
        * long_expression_2 // post commnt 2
        // pre comment 3
        * long_expression_3 // post commnt 3
        // pre comment 4
        * long_expression_4 // post commnt 4
        // pre comment 5
        * long_expression_5 // post commnt 5
        // pre comment 6
        * long_expression_6 // post commnt 6
```

given the above expression, each pre and post comment should stay as is.